file/pkg/model/fileshare.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 The SODA Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import "github.com/globalsign/mgo/bson"
+
+// FileShareS is a schema for fileshare API. Fileshare will be created on cloud backend.
+type FileShare struct {
+	// The uuid of the file share, it's unique in the context and generated by system
+	// on successful creation of the file share. It's not allowed to be modified by
+	// the user.
+	// +readOnly
+	Id bson.ObjectId `json:"id,omitempty" bson:"_id,omitempty"`
+
+	// CreatedAt representing the server time when the object was created successfully.
+	// Now, it's represented as a time string in RFC8601 format.
+	// +readOnly
+	CreatedAt string `json:"createdAt" bson:"createdAt,omitempty"`
+
+	// UpdatedAt representing the server time when the object was updated successfully.
+	// Now, it's represented as a time string in RFC8601 format.
+	// +readOnly
+	UpdatedAt string `json:"updatedAt" bson:"updatedAt,omitempty"`
+
+	// The uuid of the project that the file share belongs to.
+	TenantId string `json:"tenantId,omitempty" bson:"tenantId,omitempty"`
+
+	// The uuid of the user that the file share belongs to.
+	// +optional
+	UserId string `json:"userId,omitempty" bson:"userId,omitempty"`
+
+	// The uuid of the backend that the file share belongs to.
+	BackendId string `json:"backendId,omitempty" bson:"backendId,omitempty"`
+
+	// The name of the backend that the file share belongs to.
+	Backend string `json:"backend,omitempty" bson:"backend,omitempty"`
+
+	// The name of the file share.
+	Name string `json:"name,omitempty" bson:"name,omitempty"`
+
+	// The description of the file share.
+	// +optional
+	Description string `json:"description,omitempty" bson:"description,omitempty"`
+
+	// The type of the file share.
+	Type string `json:"type,omitempty" bson:"type,omitempty"`
+
+	// The size of the file share requested by the user.
+	// Default unit of file share Size is GB.
+	// +optional
+	Size int64 `json:"size,omitempty" bson:"size,omitempty"`
+
+	// The location that file share belongs to.
+	// +optional
+	Region string `json:"region,omitempty bson:"region,omitempty"`
```

Comment:
What's the definition of this Region?

Reply:
Region may be required for creating Mount Targets

Do you have any suggestions for Region?

Reply:
Does all cloud file share services have this kind of parameters? I'm just thinking we need to find a normal way. Maybe that depends on the analysis of all parameters for each cloud file share service. If you have such analysis, that would help.